pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -516,6 +548,60 @@ func (k *Kubernetes) ConfigTmpfs(name string, service kobject.ServiceConfig) ([]
 	return volumeMounts, volumes
 }
 
+// ConfigSecretVolumes config volumes from secret.
+// Link: https://docs.docker.com/compose/compose-file/#secrets
+// In kubernetes' Secret resource, it has a data structure like a map[string]bytes, every key will act like the file name
+// when mount to a container. This is the part that missing in compose. So we will create a single key secret from compose
+// config and the key's name will be the secret's name, it's value is the file content.
+// compose'secret can only be mounted at `/run/secrets`, so we will hardcoded this.
+func (k *Kubernetes) ConfigSecretVolumes(name string, service kobject.ServiceConfig) ([]api.VolumeMount, []api.Volume) {
+	var volumeMounts []api.VolumeMount
+	var volumes []api.Volume
+	if len(service.Secrets) > 0 {
+		for _, secretConfig := range service.Secrets {
+			if secretConfig.UID != "" {
+				log.Warnf("Ignore pid in secrets for service: %s", name)
+			}
+			if secretConfig.GID != "" {
+				log.Warnf("Ignore gid in secrets for service: %s", name)
+			}
+
+			target := secretConfig.Target
+			if target == "" {
+				target = secretConfig.Source
+			}
+
+			volSource := api.VolumeSource{
+				Secret: &api.SecretVolumeSource{
+					SecretName: secretConfig.Source,
+					Items: []api.KeyToPath{{
+						Key:  secretConfig.Source,
+						Path: target,
+					}},
+				},
+			}
+
+			if secretConfig.Mode != nil {
+				mode := cast.ToInt32(*secretConfig.Mode)
+				volSource.Secret.DefaultMode = &mode
+			}
+
+			vol := api.Volume{
+				Name:         secretConfig.Source,
+				VolumeSource: volSource,
+			}
+			volumes = append(volumes, vol)
+
+			volMount := api.VolumeMount{
+				Name:      vol.Name,
+				MountPath: "/run/secrets",
```

Comment:
@hangyan 
This shouldn't be `MountPath: "/run/secrets/" + vol.Name,` ???

Because without the MountPath targeting to the secret, it gives a error:
```
RunContainerError: failed to start container "f97d9e9354548939d15acae70758347538426fbb05c80f2a5cd316dc67e7bf58": 

Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:359: 

container init caused \"rootfs_linux.go:54: 
mounting \\\"/opt/rke/var/lib/kubelet/pods/516c135b-bce8-11e8-8714-005056ae708d/volumes/kubernetes.io~secret/default-token-khnzc\\\" 
to rootfs \\\"/var/lib/docker/overlay/60fb5d9a4007bf8e1889898df737b275cb77431351b9168ab4d629fbcdf065b5/merged\\\" 
at \\\"/var/lib/docker/overlay/60fb5d9a4007bf8e1889898df737b275cb77431351b9168ab4d629fbcdf065b5/merged/run/secrets/kubernetes.io/serviceaccount\\\" 
caused \\\"mkdir /var/lib/docker/overlay/60fb5d9a4007bf8e1889898df737b275cb77431351b9168ab4d629fbcdf065b5/merged/run/secrets/kubernetes.io: read-only file system\\\"\""
```
Just like in this issue: https://github.com/kubernetes/kubernetes/issues/65835

And concating the secret name, it work!

Reply:
With some research i discover that:

## kubernetes
```bash
kubectl create secret generic my_secret --from-literal=my_secret_key_1=my_secret_value_1 --from-literal=my_secret_key_2=my_secret_value_2
```
it create a file `/run/secrets/my_secret/my_secret_key_1` (with the content `my_secret_value_1`)

## docker-compose/swarm

- with short syntax:
```yaml
...
    secrets:
      - my_secret
secrets:
  my_secret:
    file: my_secret.txt
```
it create a file `/run/secrets/my_secret`
> **don't allow** to be compatible with kubernetes

- with long syntax:
```yaml
...
    secrets:
      - source: some_secret
        target: my_secret/my_secret_key
secrets:
  some_secret:
    file: my_secret.txt
```
it create a file `/run/secrets/my_secret/my_secret_key`
> **allow** to be compatible with kubernetes

---

I believe that this PR:
- **shouldn't allow to use short-syntax**, or if allow, to at least concat the secret name as directory (as my previous comment suggest to avoid the error).
- **only allow long-syntax if has at least inside one directory** to be compatible with kubernetes

Reply:
@jvitor83 Sorry about the late response. I have re-checked the document of kubernetes an docker. I have different opinion on this problem. kompose is meant to translate compose to kubernetes as precision as possible. Because docker mount the secret to `/run/secrets/<secret-name>`, so this path should be the same in kubernetes, not `/run/secrets/<secret-name>/<key>`. I believe the right choice is to use subPath ( you can see a example in the issues link you provide)

Reply:
I get it that _"kompose is meant to translate compose to kubernetes as precision as possible"_, but it didn't work with the k8s generated!
The generated mapping gives error (mentioned before) and the pod didn't start!

By my understanding, the generated files should work in first place to then keep the translation as precise as possible.

I guess we should remove the short syntax once it is incompatible and don't allow to get the file at the same path.

Reply:
The issue link you provide has a solution in the end:subPath. I think it's a better solution, i haven't test it out, but I think it should work

Reply:
@hangyan 
I am not understanding what you mean.

The thing is that this PR is generating:
```yaml
        volumeMounts:
        - name: secret-name
          mountPath: "/run/secrets"
```
which gives the error reported.
**The issue is solved not because it put the `subPath`, but because the directory was added to the mountPath. (I think the issue author have been mistaken)**
If the kompose generate the k8s without the mountPath with a directory (with secret using short-syntax), it will give error on start. _I have tested it with openjdk, dotnet and others. All gives the same error._

What you are suggesting to do? 
_If is to: "modify the code to add the subPath", it will continue to give the error._

I think we have to **at least put some warning** saying that secrets with short-syntax can gives error.